### ISSUE TITLE
Call to undefined method is ignored if used as a parameter to overloaded (with __call) method

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#839]
https://getpsalm.org/r/d8152ce3d3
```php
<?php
class A {
  public function __call(string $method, array $args) {
  }
}

class B {
  public function doWork():void {
    $q = new A;
    $q->workwork($this->zugzug());
  }
}
(new B)->doWork();
```
This should complain about `zugzug()` method not existing, but currently does not.

**Comment by weirdan** [comment]
:+1:
